spec/unit/facter/sensu_version_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "spec_helper"
+
+describe "Facter::Util::Fact" do
+  before {
+    Facter.clear
+  }
+
+  context "on linux" do
+    Facter.fact(:kernel).stubs(:value).returns('Linux')
+    describe 'returns sensu version when present' do
+        it do
+          File.stubs(:exists?).with('/opt/sensu/embedded/bin/sensu-client').returns(true)
+          Facter::Util::Resolution.expects(:exec).with('/opt/sensu/embedded/bin/sensu-client --version 2>&1').returns('0.23.3')
+          expect(Facter.value(:sensu_version)).to eq('0.23.3')
+        end
+    end
+
+    describe 'returns nil when sensu is not present' do
+        it do
+          File.stubs(:exists?).with('/opt/sensu/embedded/bin/sensu-client').returns(false)
+          expect(Facter.value(:sensu_version)).to be_nil
+        end
+    end
+  end
+
+  context "on non linux systems" do
```

Comment:
I believe this should read non-posix systems as the test should work on Solaris, OSX and FreeBSD.